presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -340,6 +342,24 @@ protected Type visitDereferenceExpression(DereferenceExpression node, AnalysisCo
             return rowFieldType;
         }
 
+        private void assertColumnPrefix(QualifiedName qualifiedName, Expression node)
+        {
+            // Recursively check if its prefix is a column.
+            while (qualifiedName.getPrefix().isPresent()) {
+                qualifiedName = qualifiedName.getPrefix().get();
+                List<Field> matches = tupleDescriptor.resolveFields(qualifiedName);
+                if (matches.size() > 1) {
+                    throw new SemanticException(AMBIGUOUS_ATTRIBUTE, node, "Column '%s' is ambiguous", node);
```

Comment:
This should use `qualifiedName` in the error message instead of `node` since `qualifiedName` is the column that is ambiguous